packages/syft/src/syft/client/client.py

```diff
@@ -664,40 +666,13 @@ def registries(self) -> Any:
 
     @property
     def images(self) -> Any:
-        if self.api.has_service("worker_image"):
-            return self.api.services.worker_image
-        return None
-
-    def build_image(
-        self,
-        image: UID,
-        repo_name: str = "",
-        version="latest",
-        registry: Optional[UID] = None,
-    ):
         if not self.api.has_service("worker_image"):
             return None
 
-        return self.api.services.worker_image.build(
-            image=image,
-            tag=repo_name,
-            version=version,
-            registry=registry,
-        )
+        api = self.api.services.worker_image
+        api.push = _prompt_for_password(api.push)
```

Comment:
we have to clone `self.api.services.worker_image` here somehow. so that `self.api.services.worker_image.push` is not modified, only `client.images.push`.

Reply:
maybe we can make this a bit more generic?

```python
prompt_map = {
    "push": dict(password="Enter registry password"),
    "delete": dict(password="Enter password to delete the registry"),
    "api_func": dict(arg="Enter arg to pass through to api_func"),
}
return with_prompts(self.api.services.worker_image, prompt_map )
```

Reply:
We should make it more generic later on when we need to use similar things in different places. Currently this is the only place where we need this so I think this is sufficient, best to avoid premature abstraction. Prompting for password requires using `getpass` while `input` is more suitable for other arguments so to make it work for any arbitrary arg we would need to handle that too and that would be too much for now.

Reply:
I do think it's not unlikely that we need to use more of this pattern. As I understand `client.<service>` is more suited for interactivity and `client.api.services.<service>` is more for scripting/programmable stuff and there are definitely frictions between them and we might need to adapt the latter to better fit the former. Just think that as more concrete use cases surface we would have a better idea on how to make the right abstraction.